compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -82,29 +82,24 @@ impl<'tcx, M: Machine<'tcx>> LayoutOfHelpers<'tcx> for InterpCx<'tcx, M> {
     }
 
     #[inline]
-    fn handle_layout_err(
-        &self,
-        err: LayoutError<'tcx>,
-        _: Span,
-        _: Ty<'tcx>,
-    ) -> InterpErrorInfo<'tcx> {
-        err_inval!(Layout(err)).into()
+    fn handle_layout_err(&self, err: LayoutError<'tcx>, _: Span, _: Ty<'tcx>) -> InterpError<'tcx> {
+        err_inval!(Layout(err))
     }
 }
 
 impl<'tcx, M: Machine<'tcx>> FnAbiOfHelpers<'tcx> for InterpCx<'tcx, M> {
-    type FnAbiOfResult = InterpResult<'tcx, &'tcx FnAbi<'tcx, Ty<'tcx>>>;
+    type FnAbiOfResult = Result<&'tcx FnAbi<'tcx, Ty<'tcx>>, InterpError<'tcx>>;
```

Comment:
Why is this not using InterpResult?

Reply:
Because then `InterpResult` would have to implement `MaybeResult` which could be used to bypass `discard_err`.

Also, I did the same for `LayoutOfHelpers`. That has the benefit that the calls to `ecx.layout_of()` actually don't have to bother with all this `InterpResult` stuff, they can use the usual `Result` machinery -- which is fine, we knows those will never be UB so they are harmless to discard.